### ISSUE TITLE
[$250] Scan – RBR is not displayed in LHN when editing an amount to be higher than the scanned amount

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#47411]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.20-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4856399
**Email or phone of affected tester (no customers):** htad26+dfdskffdgfh@gmail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
Prerequisite
Have a workspace and invite a member (Employee) to the workspace
1. As the employee navigate to the workspace chat
2. Click on + icon > Submit expense > Scan
3. Upload a digital receipt > Submit > Wait for the smart scan to be completed
4. Once the smart scan is finished navigate to the expense details
5. Click on the amount > edit the amount to be higher than the scanned amount > Save
6. Notice that after a while a violation is displayed indicating the amount is higher than the scanned receipt
7. Now observe the LHN of the workspace chat

## Expected Result:
RBR is displayed in the expense and LHN

## Actual Result:
RBR is not displayed in LHN

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/user-attachments/assets/24cbf938-978d-4ab6-bafe-130ca18e5ec0

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0124127bfc33770a35</li>
        <li>Upwork Job ID: 1824242806372548696</li>
        <li>Last Price Increase: 2024-08-16</li>
<li>Automatic offers: </li>
<ul>
<li>rojiphil | Reviewer | 103558418</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @rojiphil</details>

**Comment by lanitochka17** [comment]
We think that this bug might be related to #wave-collect - Release 1

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
RBR is not displayed when the receipt has an error of amount is bigger than the receipt amount.

### What is the root cause of that problem?
When we have a violation of the unmatched data with the receipt, the violation type will be a NOTICE. 
<img width="204" alt="image" src="https://github.com/user-attachments/assets/983ccbd1-2db5-4fb3-bc2b-61cf3ebdd7db">

Currently, the RBR will only show if the violation type is either a VIOLATION or a WARNING.
https://github.com/Expensify/App/blob/27e62e7f72e8f6cf8a9c57f49998d8aa307cdde5/src/libs/ReportUtils.ts#L5759

### What changes do you think we should make in order to solve the problem?

Maybe it's expected to ignore the NOTICE type violation, but if we want to change this, then we can include checking the NOTICE type violation.
```
 || TransactionUtils.hasNoticeTypeViolation(IOUTransactionID, transactionViolations)
```

Or we can be more specific to check if it's a NOTICE violation and also a `modifiedAmount` or `modifiedDate` violation name.

**Comment by mallenexpensify** [comment]
I think I was able to repro.  One thing I don't understand - After creating the expense, I see the `Expense Report` chat in LHN (cuz the report is in the main pane).  Once I click away, I don't see the `Expense Report` in LHN. 
<img width="950" alt="image" src="https://github.com/user-attachments/assets/9de347c7-6552-4556-b75d-0208577acac2">

I see this, and when I click into the report, it shows in LHN while it's open then hides once I click another chat in LHN
<img width="971" alt="image" src="https://github.com/user-attachments/assets/4dc9a243-60fd-4909-a932-0cd03be988d9">

@rojiphil can you attempt reproduction and comment on my observation above? Thx

**Comment by rojiphil** [comment]
> when I click into the report, it shows in LHN while it's open then hides once I click another chat in LHN

> can you attempt reproduction and comment on my observation above? Thx

@mallenexpensify I think it is designed to hide unless it is pinned or if there is GBR/RBR. Once we fix this issue, the RBR will be enabled which will ensure that the report remains in LHN.

**Comment by rojiphil** [comment]
@bernhardoj [proposal](https://github.com/Expensify/App/issues/47411#issuecomment-2288640345) RCA makes sense. I like the proposed solution of fixing it by checking the NOTICE violation type and `modifiedAmount`/`modifiedDate` violation name. This will additionally fix the issue with `modifiedDate`.
🎀👀🎀 C+ reviewed

**Comment by bernhardoj** [comment]
PR is ready

cc: @rojiphil

**Comment by IuliiaHerets** [comment]
This PR is failing for Desktop because of issue #48147

**Comment by bernhardoj** [comment]
I have opened a new PR to address https://github.com/Expensify/App/issues/48147.

**Comment by rojiphil** [comment]
> This PR is failing for Desktop because of issue #48147

The mentioned issue is a non-issue as it is already closed [here](https://github.com/Expensify/App/issues/48147#issuecomment-2321757131)

**Comment by bernhardoj** [comment]
Oh, shoot. I didn't realize it was closed. I believe we still need to fix it because the RBR shows in LHN but no error message is shown in the report.

**Comment by rojiphil** [comment]
> I believe we still need to fix it because the RBR shows in LHN but no error message is shown in the report.

Yeah. A BE issue though but we can make it consistent in FE. Will review soon.

**Comment by mallenexpensify** [comment]
@adelekennedy  I'm out til Tuesday, can you keep 👀 on plz? Thx

**Comment by francoisl** [comment]
This must have been on production for a while now ([staging deploy was Sept 9](https://github.com/Expensify/App/pull/48236#issuecomment-2338684837)), I think the automation for the production didn't work. @mallenexpensify let's issue the payments now and close this.

**Comment by mallenexpensify** [comment]
Contributor: @bernhardoj owed $250 via NewDot
Contributor+: @rojiphil paid $250 via Upwork. 

@rojiphil plz complete the BZ checklist.  

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by JmillsExpensify** [comment]
$250 approved for @bernhardoj

**Comment by rojiphil** [comment]
* [x]  [@rojiphil] The PR that introduced the bug has been identified. Link to the PR: [Offending PR](https://github.com/Expensify/App/pull/47094)
* [x]  [@rojiphil] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: [Added comment](https://github.com/Expensify/App/pull/47094#issuecomment-2370638863)
* [x]  [@rojiphil] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: Not Required. Existing checklist is good enough to capture such issues.
* [x]  [@rojiphil] Determine if we should create a regression test for this bug. : Yes. We can
* [x]  [@rojiphil] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. 

Steps:
	1	Submit a scan request with a receipt that has a date
	2	Wait until the smartscan is finished. Make sure there is no error with the request
	3	Edit the amount to be higher than the original amount
	4	(for small screen) Go back to LHN
	5	Verify there is an RBR on the money request report/thread
	6	Reset the amount back to the original
	7	Edit the date
	8	(for small screen) Go back to LHN
	9	Verify there is an RBR on the money request report/thread

**Comment by rojiphil** [comment]
@mallenexpensify BZ checklist done. Thanks

**Comment by mallenexpensify** [comment]
Thanks @rojiphil test case created
- https://github.com/Expensify/Expensify/issues/430817
